In [ ]:
import pandas as pd 

# `Reading the datasets`

In [ ]:
benefitial_df = pd.read_csv("./data/Train_Beneficiarydata-1542865627584.csv")
inpatient_df = pd.read_csv("./data/Train_Inpatientdata-1542865627584.csv")
outpatient_df = pd.read_csv('./data/Train_Outpatientdata-1542865627584.csv')

In [ ]:
benefitial_df.head()

BeneID         DOB  DOD  Gender  Race RenalDiseaseIndicator  State  \
0  BENE11001  1943-01-01  NaN       1     1                     0     39   
1  BENE11002  1936-09-01  NaN       2     1                     0     39   
2  BENE11003  1936-08-01  NaN       1     1                     0     52   
3  BENE11004  1922-07-01  NaN       1     1                     0     39   
4  BENE11005  1935-09-01  NaN       1     1                     0     24   

   County  NoOfMonths_PartACov  NoOfMonths_PartBCov  ...  \
0     230                   12                   12  ...   
1     280                   12                   12  ...   
2     590                   12                   12  ...   
3     270                   12                   12  ...   
4     680                   12                   12  ...   

   ChronicCond_Depression  ChronicCond_Diabetes  ChronicCond_IschemicHeart  \
0                       1                     1                          1   
1                       2                     2                          2   
2                       2                     2                          1   
3                       2                     1                          1   
4                       2                     1                          2   

   ChronicCond_Osteoporasis  ChronicCond_rheumatoidarthritis  \
0                         2                                1   
1                         2                                2   
2                         2                                2   
3                         1                                1   
4                         2                                2   

   ChronicCond_stroke  IPAnnualReimbursementAmt  IPAnnualDeductibleAmt  \
0                   1                     36000                   3204   
1                   2                         0                      0   
2                   2                         0                      0   
3                   2                         0                      0   
4                   2                         0                      0   

   OPAnnualReimbursementAmt  OPAnnualDeductibleAmt  
0                        60                     70  
1                        30                     50  
2                        90                     40  
3                      1810                    760  
4                      1790                   1200  

[5 rows x 25 columns]

In [ ]:
inpatient_df.head() 

BeneID   ClaimID ClaimStartDt  ClaimEndDt  Provider  \
0  BENE11001  CLM46614   2009-04-12  2009-04-18  PRV55912   
1  BENE11001  CLM66048   2009-08-31  2009-09-02  PRV55907   
2  BENE11001  CLM68358   2009-09-17  2009-09-20  PRV56046   
3  BENE11011  CLM38412   2009-02-14  2009-02-22  PRV52405   
4  BENE11014  CLM63689   2009-08-13  2009-08-30  PRV56614   

   InscClaimAmtReimbursed AttendingPhysician OperatingPhysician  \
0                   26000          PHY390922                NaN   
1                    5000          PHY318495          PHY318495   
2                    5000          PHY372395                NaN   
3                    5000          PHY369659          PHY392961   
4                   10000          PHY379376          PHY398258   

  OtherPhysician AdmissionDt  ... ClmDiagnosisCode_7  ClmDiagnosisCode_8  \
0            NaN  2009-04-12  ...               2724               19889   
1            NaN  2009-08-31  ...                NaN                 NaN   
2      PHY324689  2009-09-17  ...                NaN                 NaN   
3      PHY349768  2009-02-14  ...              25062               40390   
4            NaN  2009-08-13  ...               5119               29620   

  ClmDiagnosisCode_9 ClmDiagnosisCode_10 ClmProcedureCode_1  \
0               5849                 NaN                NaN   
1                NaN                 NaN             7092.0   
2                NaN                 NaN                NaN   
3               4019                 NaN              331.0   
4              20300                 NaN             3893.0   

  ClmProcedureCode_2 ClmProcedureCode_3 ClmProcedureCode_4 ClmProcedureCode_5  \
0                NaN                NaN                NaN                NaN   
1                NaN                NaN                NaN                NaN   
2                NaN                NaN                NaN                NaN   
3                NaN                NaN                NaN                NaN   
4                NaN                NaN                NaN                NaN   

  ClmProcedureCode_6  
0                NaN  
1                NaN  
2                NaN  
3                NaN  
4                NaN  

[5 rows x 30 columns]

In [ ]:
outpatient_df.head()

BeneID    ClaimID ClaimStartDt  ClaimEndDt  Provider  \
0  BENE11002  CLM624349   2009-10-11  2009-10-11  PRV56011   
1  BENE11003  CLM189947   2009-02-12  2009-02-12  PRV57610   
2  BENE11003  CLM438021   2009-06-27  2009-06-27  PRV57595   
3  BENE11004  CLM121801   2009-01-06  2009-01-06  PRV56011   
4  BENE11004  CLM150998   2009-01-22  2009-01-22  PRV56011   

   InscClaimAmtReimbursed AttendingPhysician OperatingPhysician  \
0                      30          PHY326117                NaN   
1                      80          PHY362868                NaN   
2                      10          PHY328821                NaN   
3                      40          PHY334319                NaN   
4                     200          PHY403831                NaN   

  OtherPhysician ClmDiagnosisCode_1  ... ClmDiagnosisCode_9  \
0            NaN              78943  ...                NaN   
1            NaN               6115  ...                NaN   
2            NaN               2723  ...                NaN   
3            NaN              71988  ...                NaN   
4            NaN              82382  ...                NaN   

  ClmDiagnosisCode_10 ClmProcedureCode_1 ClmProcedureCode_2  \
0                 NaN                NaN                NaN   
1                 NaN                NaN                NaN   
2                 NaN                NaN                NaN   
3                 NaN                NaN                NaN   
4                 NaN                NaN                NaN   

  ClmProcedureCode_3 ClmProcedureCode_4 ClmProcedureCode_5 ClmProcedureCode_6  \
0                NaN                NaN                NaN                NaN   
1                NaN                NaN                NaN                NaN   
2                NaN                NaN                NaN                NaN   
3                NaN                NaN                NaN                NaN   
4                NaN                NaN                NaN                NaN   

  DeductibleAmtPaid  ClmAdmitDiagnosisCode  
0                 0                  56409  
1                 0                  79380  
2                 0                    NaN  
3                 0                    NaN  
4                 0                  71947  

[5 rows x 27 columns]

***
# `Data overview`

In [ ]:
benefitial_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 138556 entries, 0 to 138555
Data columns (total 25 columns):
 #   Column                           Non-Null Count   Dtype 
---  ------                           --------------   ----- 
 0   BeneID                           138556 non-null  object
 1   DOB                              138556 non-null  object
 2   DOD                              1421 non-null    object
 3   Gender                           138556 non-null  int64 
 4   Race                             138556 non-null  int64 
 5   RenalDiseaseIndicator            138556 non-null  object
 6   State                            138556 non-null  int64 
 7   County                           138556 non-null  int64 
 8   NoOfMonths_PartACov              138556 non-null  int64 
 9   NoOfMonths_PartBCov              138556 non-null  int64 
 10  ChronicCond_Alzheimer            138556 non-null  int64 
 11  ChronicCond_Heartfailure         138556 non-null  int64 
 12  ChronicCond_Kidn

In [ ]:
inpatient_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 40474 entries, 0 to 40473
Data columns (total 30 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   BeneID                  40474 non-null  object 
 1   ClaimID                 40474 non-null  object 
 2   ClaimStartDt            40474 non-null  object 
 3   ClaimEndDt              40474 non-null  object 
 4   Provider                40474 non-null  object 
 5   InscClaimAmtReimbursed  40474 non-null  int64  
 6   AttendingPhysician      40362 non-null  object 
 7   OperatingPhysician      23830 non-null  object 
 8   OtherPhysician          4690 non-null   object 
 9   AdmissionDt             40474 non-null  object 
 10  ClmAdmitDiagnosisCode   40474 non-null  object 
 11  DeductibleAmtPaid       39575 non-null  float64
 12  DischargeDt             40474 non-null  object 
 13  DiagnosisGroupCode      40474 non-null  object 
 14  ClmDiagnosisCode_1      40474 non-null

In [ ]:
outpatient_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 517737 entries, 0 to 517736
Data columns (total 27 columns):
 #   Column                  Non-Null Count   Dtype  
---  ------                  --------------   -----  
 0   BeneID                  517737 non-null  object 
 1   ClaimID                 517737 non-null  object 
 2   ClaimStartDt            517737 non-null  object 
 3   ClaimEndDt              517737 non-null  object 
 4   Provider                517737 non-null  object 
 5   InscClaimAmtReimbursed  517737 non-null  int64  
 6   AttendingPhysician      516341 non-null  object 
 7   OperatingPhysician      90617 non-null   object 
 8   OtherPhysician          195046 non-null  object 
 9   ClmDiagnosisCode_1      507284 non-null  object 
 10  ClmDiagnosisCode_2      322357 non-null  object 
 11  ClmDiagnosisCode_3      203257 non-null  object 
 12  ClmDiagnosisCode_4      125596 non-null  object 
 13  ClmDiagnosisCode_5      74344 non-null   object 
 14  ClmDiagnosisCode_6  

***
# `Investigating the Beneficiary patients dataset`

In [ ]:
benefitial_df

BeneID         DOB  DOD  Gender  Race RenalDiseaseIndicator  \
0        BENE11001  1943-01-01  NaN       1     1                     0   
1        BENE11002  1936-09-01  NaN       2     1                     0   
2        BENE11003  1936-08-01  NaN       1     1                     0   
3        BENE11004  1922-07-01  NaN       1     1                     0   
4        BENE11005  1935-09-01  NaN       1     1                     0   
...            ...         ...  ...     ...   ...                   ...   
138551  BENE159194  1939-07-01  NaN       1     1                     0   
138552  BENE159195  1938-12-01  NaN       2     1                     0   
138553  BENE159196  1916-06-01  NaN       2     1                     0   
138554  BENE159197  1930-01-01  NaN       1     1                     0   
138555  BENE159198  1952-04-01  NaN       2     1                     0   

        State  County  NoOfMonths_PartACov  NoOfMonths_PartBCov  ...  \
0          39     230                   12                   12  ...   
1          39     280                   12                   12  ...   
2          52     590                   12                   12  ...   
3          39     270                   12                   12  ...   
4          24     680                   12                   12  ...   
...       ...     ...                  ...                  ...  ...   
138551     39     140                   12                   12  ...   
138552     49     530                   12                   12  ...   
138553      6     150                   12                   12  ...   
138554     16     560                   12                   12  ...   
138555     21      20                   12                   12  ...   

        ChronicCond_Depression  ChronicCond_Diabetes  \
0                            1                     1   
1                            2                     2   
2                            2                     2   
3                            2                     1   
4                            2                     1   
...                        ...                   ...   
138551                       2                     2   
138552                       2                     1   
138553                       1                     1   
138554                       2                     2   
138555                       1                     1   

        ChronicCond_IschemicHeart  ChronicCond_Osteoporasis  \
0                               1                         2   
1                               2                         2   
2                               1                         2   
3                               1                         1   
4                               2                         2   
...                           ...                       ...   
138551                          2                         2   
138552                          2                         2   
138553                          1                         2   
138554                          1                         2   
138555                          2                         2   

        ChronicCond_rheumatoidarthritis  ChronicCond_stroke  \
0                                     1                   1   
1                                     2                   2   
2                                     2                   2   
3                                     1                   2   
4                                     2                   2   
...                                 ...                 ...   
138551                                2                   2   
138552                                2                   2   
138553                                2                   2   
138554                                2                   2   
138555                                1                   2   

        IPAnnualReimbursementAmt  IPAnnualDeductibleAmt  \


### Checking the percentage of the NaNs

In [ ]:
# Check the percentage of Nan in dataset
total = benefitial_df.isnull().sum().sort_values(ascending=False)
percent = (benefitial_df.isnull().sum()/len(benefitial_df)).sort_values(ascending=False)
missing_data = pd.concat([total, percent], axis=1, keys=['Total', 'Percent'])
missing_data 

Total   Percent
DOD                              137135  0.989744
BeneID                                0  0.000000
ChronicCond_Cancer                    0  0.000000
OPAnnualReimbursementAmt              0  0.000000
IPAnnualDeductibleAmt                 0  0.000000
IPAnnualReimbursementAmt              0  0.000000
ChronicCond_stroke                    0  0.000000
ChronicCond_rheumatoidarthritis       0  0.000000
ChronicCond_Osteoporasis              0  0.000000
ChronicCond_IschemicHeart             0  0.000000
ChronicCond_Diabetes                  0  0.000000
ChronicCond_Depression                0  0.000000
ChronicCond_ObstrPulmonary            0  0.000000
ChronicCond_KidneyDisease             0  0.000000
DOB                                   0  0.000000
ChronicCond_Heartfailure              0  0.000000
ChronicCond_Alzheimer                 0  0.000000
NoOfMonths_PartBCov                   0  0.000000
NoOfMonths_PartACov                   0  0.000000
County                                0  0.000000
State                                 0  0.000000
RenalDiseaseIndicator                 0  0.000000
Race                                  0  0.000000
Gender                                0  0.000000
OPAnnualDeductibleAmt                 0  0.000000

### `BeneId`: Beneficiary Patient ID

In [ ]:
benefitial_df['BeneID'].dtype

dtype('O')

In [ ]:
benefitial_df['BeneID'].str.startswith('BENE').value_counts()

True    138556
Name: BeneID, dtype: int64

<div class="alert alert-block alert-success">All the <strong>BeneID</strong> values are valid and there is no missing values.
</div>

***

### `DOB`: Date of Birth

In [ ]:
benefitial_df['DOB'].dtype

dtype('O')

The date datatype is an object.

In [ ]:
#converting the date object into datetime
benefitial_df['DOB'] = pd.to_datetime(benefitial_df['DOB'])
benefitial_df['DOB'].dtype

dtype('<M8[ns]')

In [ ]:
#check if there is an abnormal date.
print("oldest date -> ",benefitial_df['DOB'].min(),"\nnewest date -> ",benefitial_df['DOB'].max())

oldest date ->  1909-01-01 00:00:00 
newest date ->  1983-12-01 00:00:00


In [ ]:
pd.DatetimeIndex(benefitial_df['DOB']).month.value_counts()

12    11731
9     11656
3     11648
10    11646
4     11625
5     11579
11    11569
7     11513
8     11473
2     11407
6     11361
1     11348
Name: DOB, dtype: int64

In [ ]:
pd.DatetimeIndex(benefitial_df['DOB']).day.value_counts()

1    138556
Name: DOB, dtype: int64

<div class="alert alert-block alert-warning">We noticed that all the <strong>DOB</strong> days have one value which is 1. We assume that they ignore the day and only register the month and the year.
</div>

<div class="alert alert-block alert-success">All the <strong>DOB</strong> values are valid and there is no missing values.
</div>

***

### `DOD`: Date of Death

In [ ]:
benefitial_df['DOD'].isna().value_counts()/len(benefitial_df)

True     0.989744
False    0.010256
Name: DOD, dtype: float64

98% of the DOD are missing because not all patients are dead yet. As a result, we will check the dates of the dead people.

In [ ]:
benefitial_df['DOD'].dtype

dtype('O')

The date datatype is an object.

In [ ]:
#converting the date object into datetime
benefitial_df['DOD'] = pd.to_datetime(benefitial_df['DOD'])
benefitial_df['DOD'].dtype

dtype('<M8[ns]')

In [ ]:
#check if there is an abnormal date.
print("oldest date -> ",benefitial_df['DOD'].min(),"\nnewest date -> ",benefitial_df['DOD'].max())

oldest date ->  2009-02-01 00:00:00 
newest date ->  2009-12-01 00:00:00


In [ ]:
pd.DatetimeIndex(benefitial_df['DOD']).month.value_counts()

12.0    182
10.0    168
9.0     164
11.0    149
8.0     144
7.0     141
5.0     119
6.0     119
4.0      94
3.0      91
2.0      50
Name: DOD, dtype: int64

In [ ]:
pd.DatetimeIndex(benefitial_df['DOD']).day.value_counts()

1.0    1421
Name: DOD, dtype: int64

<div class="alert alert-block alert-warning">We noticed that all the <strong>DOD</strong> days have one value which is 1. We assume that they ignore the day and only register the month and the year.
</div>

<div class="alert alert-block alert-danger">98% of the <strong>DOD</strong> values are missing.
</div>

***

# `Gender`

In [ ]:
benefitial_df['Gender'].dtype

dtype('int64')

In [ ]:
#since the gender is a categorical data, we will convert it into category.
benefitial_df['Gender'] = benefitial_df['Gender'].astype('category')

In [ ]:
benefitial_df['Gender'].dtype

CategoricalDtype(categories=[1, 2], ordered=False)

In [ ]:
benefitial_df['Gender'].value_counts()

2    79106
1    59450
Name: Gender, dtype: int64

<div class="alert alert-block alert-success">There is no missing data in <strong>Gender</strong>.
</div>

***

# `Race`

In [ ]:
benefitial_df['Race'].dtype

dtype('int64')

In [ ]:
benefitial_df['Race'].value_counts()

1    117057
2     13538
3      5059
5      2902
Name: Race, dtype: int64

In [ ]:
#since the race is a categorical data, we will convert it into category.
benefitial_df['Race'] = benefitial_df['Race'].astype('category')

In [ ]:
benefitial_df['Race'].dtype

CategoricalDtype(categories=[1, 2, 3, 5], ordered=False)

<div class="alert alert-block alert-success">There is no missing data in <strong>Race</strong>.
</div>

***

# `RenalDiseaseIndicator`

In [ ]:
benefitial_df['RenalDiseaseIndicator'].dtype

dtype('O')

In [ ]:
benefitial_df['RenalDiseaseIndicator'].value_counts()

0    118978
Y     19578
Name: RenalDiseaseIndicator, dtype: int64

In [ ]:
#since the RenalDiseaseIndicator is a categorical data, we will convert it into category.
benefitial_df['RenalDiseaseIndicator'] = benefitial_df['RenalDiseaseIndicator'].astype('category')

In [ ]:
benefitial_df['RenalDiseaseIndicator'].dtype

CategoricalDtype(categories=['0', 'Y'], ordered=False)

<div class="alert alert-block alert-warning">We noticed that the values are (0 any Y) not (N or Y) or (0 or 1). For now, we assume that 0 means that the patient does not have a renal disease and Y is the opposite.
</div>

<div class="alert alert-block alert-success">There is no missing data in <strong>RenalDiseaseIndicaton</strong>.
</div>

***

# `State`

In [ ]:
benefitial_df['State'].dtype

dtype('int64')

In [ ]:
benefitial_df['State'].value_counts()

5     12052
10     9771
45     8780
33     8443
39     6055
14     5923
36     5366
23     5293
34     4629
31     4124
11     3892
49     3623
44     3390
22     3306
15     3272
26     3108
50     2793
52     2662
1      2615
21     2570
42     2524
18     2519
3      2395
24     2165
19     1998
37     1926
4      1817
6      1807
16     1746
25     1622
7      1602
38     1468
17     1339
54     1237
51     1212
20      931
28      905
32      895
46      851
29      752
30      732
13      687
27      538
12      484
43      465
41      438
8       415
47      410
35      304
53      295
9       214
2       196
Name: State, dtype: int64

In [ ]:
#since the State is a categorical data, we will convert it into category.
benefitial_df['State'] = benefitial_df['State'].astype('category')

In [ ]:
benefitial_df['State'].dtype

CategoricalDtype(categories=[ 1,  2,  3,  4,  5,  6,  7,  8,  9, 10, 11, 12, 13, 14, 15,
                  16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30,
                  31, 32, 33, 34, 35, 36, 37, 38, 39, 41, 42, 43, 44, 45, 46,
                  47, 49, 50, 51, 52, 53, 54],
, ordered=False)

<div class="alert alert-block alert-success">There is no missing data in <strong>State</strong>.
</div>

***

# `County`

In [ ]:
benefitial_df['County'].dtype

dtype('int64')

In [ ]:
benefitial_df['County'].value_counts()

200    3943
10     3587
20     3176
60     3003
0      2977
       ... 
944       1
742       1
753       1
744       1
521       1
Name: County, Length: 314, dtype: int64

In [ ]:
#since the County is a categorical data, we will convert it into category.
benefitial_df['County'] = benefitial_df['County'].astype('category')

In [ ]:
benefitial_df['County'].dtype

CategoricalDtype(categories=[  0,   1,  10,  11,  14,  20,  25,  30,  34,  40,
                  ...
                  987, 988, 989, 990, 991, 992, 993, 994, 996, 999],
, ordered=False)

<div class="alert alert-block alert-info">The <strong>County</strong> values range between 0 and 999.
</div>

<div class="alert alert-block alert-success">There is no missing data in <strong>State</strong>.
</div>

***

# `NoOfMonths_PartACov` + `NoOfMonths_PartBCov`

In [ ]:
benefitial_df['NoOfMonths_PartACov'].dtype

dtype('int64')

In [ ]:
benefitial_df['NoOfMonths_PartACov'].value_counts()

12    137389
0       1000
6         38
11        28
8         26
10        18
7         16
4         13
5          8
9          7
2          5
3          5
1          3
Name: NoOfMonths_PartACov, dtype: int64

In [ ]:
benefitial_df['NoOfMonths_PartBCov'].dtype

dtype('int64')

In [ ]:
benefitial_df['NoOfMonths_PartBCov'].value_counts()

12    136902
0        675
6        282
10       150
11       143
9        122
8         71
7         63
5         50
4         35
3         27
2         19
1         17
Name: NoOfMonths_PartBCov, dtype: int64

<div class="alert alert-block alert-success">There is no missing data in <strong>NoOfMonths_PartACov</strong> or <strong>NoOfMonths_PartBCov</strong>.
</div>

***

# `Choronic Diseases`

In [ ]:
choronics = ['ChronicCond_Alzheimer', 'ChronicCond_Heartfailure','ChronicCond_KidneyDisease', 'ChronicCond_Cancer','ChronicCond_ObstrPulmonary', 'ChronicCond_Depression','ChronicCond_Diabetes', 'ChronicCond_IschemicHeart','ChronicCond_Osteoporasis', 'ChronicCond_rheumatoidarthritis','ChronicCond_stroke']

In [ ]:
for disease in choronics:
    print(disease)
    print(benefitial_df[disease].dtype)
    print(benefitial_df[disease].value_counts())
    print('*********************************')

ChronicCond_Alzheimer
int64
2    92530
1    46026
Name: ChronicCond_Alzheimer, dtype: int64
*********************************
ChronicCond_Heartfailure
int64
2    70154
1    68402
Name: ChronicCond_Heartfailure, dtype: int64
*********************************
ChronicCond_KidneyDisease
int64
2    95277
1    43279
Name: ChronicCond_KidneyDisease, dtype: int64
*********************************
ChronicCond_Cancer
int64
2    121935
1     16621
Name: ChronicCond_Cancer, dtype: int64
*********************************
ChronicCond_ObstrPulmonary
int64
2    105697
1     32859
Name: ChronicCond_ObstrPulmonary, dtype: int64
*********************************
ChronicCond_Depression
int64
2    89296
1    49260
Name: ChronicCond_Depression, dtype: int64
*********************************
ChronicCond_Diabetes
int64
1    83391
2    55165
Name: ChronicCond_Diabetes, dtype: int64
*********************************
ChronicCond_IschemicHeart
int64
1    93644
2    44912
Name: ChronicCond_IschemicHeart, dtype: in

<div class="alert alert-block alert-info">All the values are either 1 or 2. We will convert the data type into categorical.
</div>

In [ ]:
#since the ChoronicDisease* are categorical, we'll convert their dtype into categorical
for disease in choronics:
    benefitial_df[disease] = benefitial_df[disease].astype('category')

<div class="alert alert-block alert-success">There is no missing data in the <strong>Choronic diseases</strong>.
</div>

***

***

# `IPAnnualReimbursementAmt` + `IPAnnualDeductibleAmt`

In [ ]:
benefitial_df['IPAnnualReimbursementAmt'].value_counts()

0        102511
4000       2123
5000       1851
3000       1800
6000       1589
          ...  
15290         1
60320         1
35500         1
39470         1
67040         1
Name: IPAnnualReimbursementAmt, Length: 3004, dtype: int64

In [ ]:
print('min: ',benefitial_df['IPAnnualReimbursementAmt'].min(),', max: ',benefitial_df['IPAnnualReimbursementAmt'].max())

min:  -8000 , max:  161470


In [ ]:
benefitial_df[benefitial_df['IPAnnualReimbursementAmt']<0]

BeneID        DOB DOD Gender Race RenalDiseaseIndicator State  \
3559     BENE14791 1929-04-01 NaT      2    3                     0    12   
8008     BENE19571 1938-07-01 NaT      2    1                     0    32   
12999    BENE24913 1946-04-01 NaT      1    1                     0    49   
15160    BENE27231 1935-02-01 NaT      1    1                     0    42   
23146    BENE35786 1937-10-01 NaT      1    1                     0     3   
31067    BENE44298 1929-02-01 NaT      2    2                     0     1   
34829    BENE48336 1941-11-01 NaT      2    1                     0    18   
40986    BENE54894 1926-04-01 NaT      2    1                     0    14   
69876    BENE85807 1972-12-01 NaT      1    1                     0    44   
75614    BENE91939 1973-09-01 NaT      2    2                     0    21   
86500   BENE103563 1934-07-01 NaT      1    1                     0    45   
88111   BENE105279 1931-11-01 NaT      2    1                     0    34   
93328   BENE110853 1913-06-01 NaT      2    3                     0     5   
113459  BENE132384 1969-08-01 NaT      1    1                     0     5   
116302  BENE135425 1924-12-01 NaT      2    1                     0    33   

       County  NoOfMonths_PartACov  NoOfMonths_PartBCov  ...  \
3559       20                   12                   12  ...   
8008      230                   12                   12  ...   
12999     930                   12                   12  ...   
15160     250                   12                   12  ...   
23146      60                   12                   12  ...   
31067     510                   12                   12  ...   
34829      70                   12                   12  ...   
40986     141                   12                   12  ...   
69876      60                   12                   12  ...   
75614      20                   12                   12  ...   
86500     810                   12                   12  ...   
88111     800                   12                   12  ...   
93328     650                   12                   12  ...   
113459    470                   12                   12  ...   
116302    420                   12                   12  ...   

       ChronicCond_Depression ChronicCond_Diabetes ChronicCond_IschemicHeart  \
3559                        2                    2                         2   
8008                        1                    1                         1   
12999                       2                    1                         1   
15160                       1                    1                         1   
23146                       1                    1                         2   
31067                       2                    2                         1   
34829                       2                    1                         1   
40986                       1                    1                         2   
69876                       1                    2                         1   
75614                       1                    1                         2   
86500                       1                    1                         1   
88111                       2                    1                         2   
93328                       1                    1                         1   
113459                      2                    2                         2   
116302                      1                    2                         2   

       ChronicCond_Osteoporasis ChronicCond_rheumatoidarthritis  \
3559                          2                               2   
8008                          2                               2   
12999                         2                               2   
15160                         1                               2   
23146                         2                               1   
31067                         2                               2   
3482

<div class="alert alert-block alert-warning">There are minus values in <strong>IPAnnualReimbursementAmt</strong>. They could mean that the hospital returns this ammount to them.
</div>

In [ ]:
benefitial_df['IPAnnualDeductibleAmt'].value_counts()

0        102019
1068      27113
2136       6418
3204       1481
4272        369
          ...  
10272         1
21204         1
1118          1
20136         1
6140          1
Name: IPAnnualDeductibleAmt, Length: 147, dtype: int64

In [ ]:
print('min: ',benefitial_df['IPAnnualDeductibleAmt'].min(),', max: ',benefitial_df['IPAnnualDeductibleAmt'].max())

min:  0 , max:  38272


<div class="alert alert-block alert-success">There is no missing data in the <strong>IPAnnualDeductibleAmt</strong>.
</div>

***

# `OPAnnualReimbursementAmt` + `OPAnnualDeductibleAmt`

In [ ]:
benefitial_df['OPAnnualReimbursementAmt'].value_counts()

0        4205
100      3916
200      3153
60       2694
300      2280
         ... 
17590       1
9530        1
21940       1
10170       1
20470       1
Name: OPAnnualReimbursementAmt, Length: 2078, dtype: int64

In [ ]:
print('min: ',benefitial_df['OPAnnualReimbursementAmt'].min(),', max: ',benefitial_df['OPAnnualReimbursementAmt'].max())

min:  -70 , max:  102960


In [ ]:
benefitial_df[benefitial_df['OPAnnualReimbursementAmt']<0]

BeneID        DOB DOD Gender Race RenalDiseaseIndicator State  \
5707     BENE17102 1929-08-01 NaT      1    1                     0    10   
17371    BENE29608 1928-12-01 NaT      2    1                     0    23   
24167    BENE36876 1935-11-01 NaT      1    1                     Y    11   
26868    BENE39788 1939-06-01 NaT      1    1                     0     7   
37622    BENE51322 1957-07-01 NaT      1    2                     0    42   
38449    BENE52207 1933-12-01 NaT      2    2                     Y    14   
48841    BENE63308 1918-06-01 NaT      2    1                     Y    14   
51826    BENE66510 1926-05-01 NaT      2    1                     0    44   
86393   BENE103447 1941-05-01 NaT      2    1                     0    21   
113093  BENE131989 1924-02-01 NaT      2    1                     0    18   
131151  BENE151251 1966-05-01 NaT      2    2                     0     7   
137402  BENE157969 1909-10-01 NaT      2    1                     0    39   

       County  NoOfMonths_PartACov  NoOfMonths_PartBCov  ...  \
5707      120                   12                   12  ...   
17371     710                   12                   12  ...   
24167     772                   12                   12  ...   
26868      10                   12                   12  ...   
37622     110                   12                   12  ...   
38449     560                   12                   12  ...   
48841     989                   12                   12  ...   
51826     930                   12                   12  ...   
86393      30                   12                   12  ...   
113093    330                   12                   12  ...   
131151     30                   12                   12  ...   
137402    290                   12                   12  ...   

       ChronicCond_Depression ChronicCond_Diabetes ChronicCond_IschemicHeart  \
5707                        2                    1                         1   
17371                       2                    1                         1   
24167                       1                    1                         1   
26868                       2                    2                         1   
37622                       2                    1                         1   
38449                       2                    1                         1   
48841                       1                    1                         1   
51826                       2                    2                         2   
86393                       2                    2                         1   
113093                      2                    2                         2   
131151                      2                    2                         1   
137402                      2                    1                         1   

       ChronicCond_Osteoporasis ChronicCond_rheumatoidarthritis  \
5707                          2                               1   
17371                         2                               2   
24167                         1                               1   
26868                         2                               2   
37622                         1                               1   
38449                         1                               2   
48841                         2                               1   
51826                         2                               2   
86393                         1                               2   
113093                        2                               2   
131151                        2                               2   
137402                        2                               2   

       ChronicCond_stroke IPAnnualReimbursementAmt IPAnnualDeductibleAmt  \
5707                    2                    44800                  2136   
17371                   2                     4400                  2136   
24167                   2           

<div class="alert alert-block alert-warning">There are minus values in <strong>OPAnnualReimbursementAmt</strong>. They could mean that the hospital returns this ammount to them.
</div>

In [ ]:
benefitial_df['OPAnnualDeductibleAmt'].value_counts()

0        13890
20        7271
10        6140
30        4755
100       4743
         ...  
8880         1
7520         1
7600         1
10080        1
4890         1
Name: OPAnnualDeductibleAmt, Length: 789, dtype: int64

In [ ]:
print('min: ',benefitial_df['OPAnnualDeductibleAmt'].min(),', max: ',benefitial_df['OPAnnualDeductibleAmt'].max())

min:  0 , max:  13840


<div class="alert alert-block alert-success">There is no missing data in the <strong>OPAnnualDeductibleAmt</strong>.
</div>

***